core/templates/pages/library-page/library-footer/library-footer.component.spec.ts

```diff
@@ -16,40 +16,72 @@
  * @fileoverview Unit tests for libraryFooter.
  */
 
-describe('Library Footer Component', function() {
-  var ctrl = null;
-  var $window = null;
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import { LibraryFooterComponent } from './library-footer.component';
+import { SharedComponentsModule } from './../../../../templates/components/shared-component.module';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { HttpClientModule } from '@angular/common/http';
 
-  var mockWindow = {
+
+class MockWindowRef {
+  _window = {
     location: {
-      pathname: ''
+      _pathname: '/math/ratios',
+      _href: '',
+      get pathname(): string {
+        return this._pathname;
+      },
+      set pathname(val: string) {
+        this._pathname = val;
+      },
     }
   };
+  get nativeWindow() {
+    return this._window;
+  }
+}
 
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    $provide.value('$window', mockWindow);
-  }));
-
-  beforeEach(angular.mock.inject(function($injector, $componentController) {
-    $window = $injector.get('$window');
-
-    ctrl = $componentController('libraryFooter', {
-      $window: $window
-    });
-  }));
+describe('Library Footer Component', () => {
+  let windowRef: MockWindowRef;
+  let fixture: ComponentFixture<LibraryFooterComponent>;
+  let component: LibraryFooterComponent;
 
-  it('should show library footer when not searching for explorations',
-    function() {
-      mockWindow.location.pathname = '/community-library';
-      ctrl.$onInit();
+  beforeEach(() => {
```

Comment:
Last nit: Can you enclose the callback with `waitForAsync`?

Reply:
> Last nit: Can you enclose the callback with `waitForAsync`?

Done